### PR TITLE
feat(wallet): add transaction review screen for ETHSwap

### PR DESCRIPTION
**Pull request by onyb**
Resolves https://github.com/brave/brave-browser/issues/24860

Slight design variations from the Figma spec were done after internal discussions with product/design.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Demo:

https://user-images.githubusercontent.com/3684187/186016130-02361dec-8a20-40e2-87b9-86a39d20ccf5.mov

## Test plan

Create a swap transaction for the following cases, and it should accurately display the maker/taker assets and amounts involved in the swap as seen in the demo above:

**Networks:** (mainnet only)
- Ethereum
- Polygon
- BSC
- Optimism
- Celo (not verified)
- Fantom (not verified)
- Avalanche (not verified)

**Pairs:**
- `ETH  -> ERC20`
- `ERC20 -> ETH`
- `ERC20 -> ERC20`

Verify in the screenshot that the correct asset logos and the corresponding network are appropriately displayed.

### CODE REVIEW

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/swap.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
+import * as React from 'react'
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/extension/confirm-transaction-panel/swap.tsx:1]: License!

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/swap.tsx [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    transactionsNetwork,
+    fromOrb,
+    toOrb,
+    updateUnapprovedTransactionNonce
+  } = usePendingTransactions()
+
+  // Memos
+  const makerAssetNetwork = React.useMemo(() => {
+    return transactionDetails?.sellToken && getTokensNetwork(networks, transactionDetails.sellToken)
+  }, [networks])
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/extension/confirm-transaction-panel/swap.tsx:87]: Missing `transactionDetails` dependency!

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/swap.tsx [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+  const takerAssetNetwork = React.useMemo(() => {
+    return transactionDetails?.buyToken && getTokensNetwork(networks, transactionDetails.buyToken)
+  }, [networks])
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/extension/confirm-transaction-panel/swap.tsx:90]: Missing `transactionDetails` dependency!

**Review by Douglashdaniel — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/common/style.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import styled from 'styled-components'
+
+import { LoaderIcon } from 'brave-ui/components/icons'
+import { WalletButton } from '../../../shared/style'
+
+interface StyleProps {
+  needsMargin: boolean
+}
```
[the remark this answers is not in the document]
Comment (onyb) [reply components/brave_wallet_ui/components/extension/confirm-transaction-panel/common/style.tsx]: Good idea! Fixed it.

**Review by Douglashdaniel — APPROVED** [review]
LGTM!

**Pull request merged**